utils/utils.go

```diff
@@ -585,3 +591,123 @@ func GetWLOLicenseAnnotations() map[string]string {
 	annotations["productChargedContainers"] = "ALL"
 	return annotations
 }
+
+func isVolumeMountFound(pts *corev1.PodTemplateSpec, name string) bool {
+	for _, v := range pts.Spec.Containers[0].VolumeMounts {
+		if v.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+func isVolumeFound(pts *corev1.PodTemplateSpec, name string) bool {
+	for _, v := range pts.Spec.Volumes {
+		if v.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+// ConfigureLTPA setups the shared-storage for LTPA keys file generation
+func ConfigureLTPA(pts *corev1.PodTemplateSpec, la *wlv1.WebSphereLibertyApplication, operatorShortName string) {
+	// Mount a volume /config/ltpa to store the ltpa.keys file
+	ltpaKeyVolumeMount := GetLTPAKeysVolumeMount(la, ltpaKeysFileName)
+	if !isVolumeMountFound(pts, ltpaKeyVolumeMount.Name) {
+		pts.Spec.Containers[0].VolumeMounts = append(pts.Spec.Containers[0].VolumeMounts, ltpaKeyVolumeMount)
+	}
+	if !isVolumeFound(pts, ltpaKeyVolumeMount.Name) {
+		vol := corev1.Volume{
+			Name: ltpaKeyVolumeMount.Name,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: operatorShortName + "-managed-ltpa",
+					Items: []corev1.KeyToPath{{
+						Key:  ltpaKeysFileName,
+						Path: ltpaKeysFileName,
+					}},
+				},
+			},
+		}
+		pts.Spec.Volumes = append(pts.Spec.Volumes, vol)
+	}
+
+	// Mount a volume /config/configDropins/overrides/ltpa.xml to store the Liberty Server XML
+	ltpaXMLVolumeMount := GetLTPAXMLVolumeMount(la, ltpaXMLFileName)
+	if !isVolumeMountFound(pts, ltpaXMLVolumeMount.Name) {
+		pts.Spec.Containers[0].VolumeMounts = append(pts.Spec.Containers[0].VolumeMounts, ltpaXMLVolumeMount)
+	}
+	if !isVolumeFound(pts, ltpaXMLVolumeMount.Name) {
+		vol := corev1.Volume{
+			Name: ltpaXMLVolumeMount.Name,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: operatorShortName + LTPAServerXMLSuffix,
+					Items: []corev1.KeyToPath{{
+						Key:  ltpaXMLFileName,
+						Path: ltpaXMLFileName,
+					}},
+				},
+			},
+		}
+		pts.Spec.Volumes = append(pts.Spec.Volumes, vol)
+	}
+}
+
+func CustomizeLTPAServerXML(xmlSecret *corev1.Secret, la *wlv1.WebSphereLibertyApplication, encryptedPassword string) {
+	xmlSecret.StringData = make(map[string]string)
+	keysFileName := strings.Replace(ltpaKeysMountPath, "/config", "${server.config.dir}", 1)
```

Comment:
Since the file is mounted to `ltpaKeysMountPath`, it's probably better to use that location as is.

Reply:
For some reason the Liberty implementation for parsing `<ltpa keysFileName="?"></ltpa>` won't do a deep folder traversal for the symlink `/config`, so I had to change it to `${server.config.dir}` for it to read the file properly.